### PR TITLE
Replace lxmenu-data with  lxqt-menu-data

### PR DESCRIPTION
**Pull request by stefonarch**
Not really sure if this is done right, thought about opening an issue but we can start fixing wrong stuff here.

### CODE REVIEW

**Comment by isf63** [comment]
On my install it says the .menu files in /etc/xdg/menus/ and the .directory files in /usr/share/desktop-directories/ are owned by lxqt-panel/lxqt-config, thus installing lxqt-menu-data fails as it's unable to overwrite them.

**Comment by stefonarch** [comment]
`lxqt-panel-git` and `lxqt-config` should be updated first, those files should be removed then

**Comment by tsujan** [comment]
> ...thus installing lxqt-menu-data fails as it's unable to overwrite them.

@yan12125, won't this cause a problem for users? The next version of `lxqt-config` will require `lxqt-menu-data`, while the latter is in conflict with the installed version of the former.

> lxqt-panel-git and lxqt-config should be updated first

 But they will need `lxqt-menu-data` (in the cmake files).

**Comment by stefonarch** [comment]
Maybe just use different names? `LXQt-Applications*.menu` and `LXQt-*-desktop.directories` ?

**Comment by yan12125** [comment]
I think there will be no conflict if lxqt-config-git, lxqt-panel-git and lxqt-menu-data-git are installed/updated at the same time. @isf63 could you paste your command and logs?

**Comment by tsujan** [comment]
> I think there will be no conflict if lxqt-config-git, lxqt-panel-git and lxqt-menu-data-git are installed/updated at the same time.

But I need to build and install `lxqt-menu-data-git`in order to build `lxqt-config-git` and `lxqt-panel-git`, while it has conflict with them. Personally, I have no problem, but it might be confusing for many.

> Maybe just use different names?

That's possible, by making `lxqt-panel` and `lxqt-config` use the new files.

**Comment by yan12125** [comment]
Oh I think I understand the situation. Usually I build packages in chroots, so I don't need to install lxqt-menu-data-git on the main system first to build lxqt-config-git and lxqt-panel-git. It's indeed tricky.

> Maybe just use different names?

Not a bad idea.

**Comment by tsujan** [comment]
The files that cause the conflicts are

```
lxqt-applications.menu
lxqt-applications-compact.menu
lxqt-applications-simple.menu
lxqt-config.menu
```

After giving it more thought, changing "applications" to "Applications" doesn't seem right to me. Other DEs don't do that. In `/etc/xdg/menus` I also have `applications.menu` belonging to kservice and `e-applications.menu` belonging to enlightenment. And please don't forget that `libfm-qt` needs to rely on `X-applications.menu` for compatibility with other DEs.

Therefore, IMHO, renaming will be problematic.

@yan12125 
If the simultaneous installation from distro repositories can be forced (can it?) and if doesn't cause any problem (does it?), then more advanced users who make their own packages could force overwriting; I think all package managers have an option for it.

**Comment by stefonarch** [comment]
My idea was only renaming the prefix to `LXQt-`, and also the `*.menu` files are conflicting except `lxqt-applications-fm.menu` as it's new.
But for the 4 desktop directory files it's not easy, rename all to make it consistent? Find another name?

Force-overwrite on arch isn't easy anymore afaik.

**Comment by stefonarch** [comment]
>Temporarily uninstalling is the safest way.

In fact usually I use `pacman -Rdd` <package>` when files are conflicting.

**Comment by yan12125** [comment]
> If the simultaneous installation from distro repositories can be forced (can it?) and if it doesn't cause any problem (does it?), then more advanced users who make their own packages could force overwriting; I think all package managers have an option for it.

Thinking again, source-based distributions like Gentoo may still have an issue like AUR. I know little about Gentoo so I cannot give a conclusion.

> Therefore, IMHO, renaming will be problematic.

> Temporarily uninstalling is the safest way.

I agree both.

**Comment by Chiitoo** [comment]
> Thinking again, source-based distributions like Gentoo may still have an issue like AUR. I know little about Gentoo so I cannot give a conclusion.

I haven't looked too deeply yet on this change, but on Gentoo two packages installing the same file can not be installed (or rather Portage will throw an error).

Depending on the situation, we can make one package block another to make sure the other is uninstalled first, or in more severe situations, where the two packages should be able to co-exist, we can split the conflicting files into their own package, and remove them from the real packages, making them depend on the "package-name-common" for example.

In other words, we have ways around conflicts.  : ]

**Comment by stefonarch** [comment]
>  I haven't looked too deeply yet on this change, but on Gentoo two packages installing the same file can not be installed (or rather Portage will throw an error).

This is not the case.  `lxqt-menu-data` provides among the replacement for lxmenu-data 8 files which previously were parts of  `lxqt-panel` and `lxqt-config`.

**Comment by Chiitoo** [comment]
Should be no problem then at all, aside from making sure they install in the correct order.

It is nice to see this dependency finally going away.

Many thanks!

**Comment by yan12125** [comment]
Thanks a lot for Gentoo information. It is great to hear that things are expected to work on Gentoo.

> It is nice to see this dependency finally going away.

lxmenu-data is not completely gone for now - see discussions about libfm-qt and pcmanfm-qt in https://github.com/lxqt/lxqt/issues/1679.

Rgarding this PR: sounds like the status quo is the best we can achieve. I left a comment at https://aur.archlinux.org/packages/lxqt-menu-data-git. Let's merge this?

**Comment by tsujan** [comment]
> lxmenu-data is not completely gone for now

Correct, but it'll be easy to remove it from `libfm-qt` too — I'll do it soon — then it'll be completely gone :)

> Let's merge this?

I agree.

**Comment by stefonarch** [comment]
Sure that I did this submodule thing correctly?

**Comment by stefonarch** [comment]
And I was sure I had `../` ... thanks!

**Comment by stefonarch** [comment]
I ran a test too, but without building, looked ok

**Comment by isf63** [comment]
I removed the lxqt-panel and lxqt-config packages and installed lxqt-menu-data, then reinstalled those two with no issues. However when trying to remove lxmenu-data my package manager said pcmanfm-qt depends on it, although it could be a needless dependency in the PKGBUILD

**Comment by stefonarch** [comment]
There was a comment from @isf63  I can't find here (received a mail and answered by mail for that): @yan12125 the AUR libfm-qt package needs apparently
 removing of` lxmenu-data`

**Comment by isf63** [comment]
I deleted my comment due to seeing in https://github.com/lxqt/lxqt/issues/1679 that @tsujan said

> pcmanfm-qt is condemned to use lxmenu-data, because of menu-cache.

It was in regards to attempting to remove the lxmenu-data package, which on my install has one "Required by", pcmanfm-qt.

**Comment by yan12125** [comment]
> pcmanfm-qt is condemned to use lxmenu-data, because of menu-cache.

**Comment by tsujan** [comment]
The git source of `libfm-qt` uses and depends on `lxqt-menu-data` now (after merging https://github.com/lxqt/libfm-qt/pull/905).

> ... which on my install has one "Required by", pcmanfm-qt.

The dependency of pcmanfm-qt on `lxmenu-data` wasn't in its source but in its package, and it was a good decision before the recent changes. But now, it can be removed.

**Comment by tsujan** [comment]
BTW, I'll apply the change to the Qt6 branch of libfm-qt soon.

**Comment by stefonarch** [comment]
I was wrong, the AUR package for pcmanfm-qt requires `lxmenu-data` https://aur.archlinux.org/packages/pcmanfm-qt-git

**Comment by Chiitoo** [comment]
Already set up the Gentoo live ebuilds (git master) and packaged lxqt-menu-data a few days ago and things seem to be working well.

Since the paths/names are different, it was very easy to deal with on the packaging side.

Many thanks!

**Comment by yan12125** [comment]
Thanks for clarification. https://aur.archlinux.org/packages/pcmanfm-qt-git is updated.
